cmd/lifecycle/analyzer.go

```diff
@@ -224,3 +223,21 @@ func (a *analyzeCmd) platformAPIVersionGreaterThan06() bool {
 func (a *analyzeCmd) restoresLayerMetadata() bool {
 	return !a.platformAPIVersionGreaterThan06()
 }
+
+func (aa *analyzeArgs) ReadableRegistryImages() []string {
+	var readableImages []string
+	if !aa.useDaemon {
+		readableImages = append(readableImages, aa.previousImageRef, aa.runImageRef)
+	}
+	return readableImages
+}
+func (aa *analyzeArgs) WriteableRegistryImages() []string {
+	var writeableImages []string
+	if aa.cacheImageRef != "" {
+		writeableImages = append(writeableImages, aa.cacheImageRef)
+	}
```

Comment:
Why do you only check that this variable is an empty string?
Unless I'm missing something you could either check that every value that you add to the `writableImages` array isn't an empty string (and the same for the `readableImages` array in `ReadableRegistryImages`) and delete the empty string condition in `verifyReadAccess` and `verifyReadWriteAccess` (in `image.go`), or just add the `aa.cacheImageRef` even when it's an empty string.

Reply:
I'm sorry, I've totally missed [this conversation](https://github.com/buildpacks/lifecycle/pull/646/files#r668830740) 😊

Reply:
Relevant: https://github.com/buildpacks/lifecycle/pull/646#discussion_r668830740

Reply:
Fixed in the following [commit](https://github.com/buildpacks/lifecycle/pull/646/commits/8a82acf4f285ed83dbfb346b7f65f106b548d8aa)